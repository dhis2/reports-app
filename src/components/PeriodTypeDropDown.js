import React from 'react'
import PropTypes from 'prop-types'
import { DropDown } from '@dhis2/d2-ui-core'
import i18n from '@dhis2/d2-i18n'

const labelText = i18n.t('Select Period Type')

function PeriodTypeDropDown({ loading, ...props }) {
    if (loading) {
        return (
            <span>
                {i18n.t('Loading period types dropdown')}
                <style jsx>{`
                    span {
                        color: #f44336;
                    }
                `}</style>
            </span>
        )
    }

    return (
        <DropDown
            fullWidth
            emptyLabel={labelText}
            hintText={labelText}
            {...props}
        />
    )
}

PeriodTypeDropDown.propTypes = {
    loading: PropTypes.bool.isRequired,
}

export default PeriodTypeDropDown
