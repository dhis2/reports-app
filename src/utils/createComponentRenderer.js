import React from 'react'
import { shallow } from 'enzyme'

export default function createComponentRenderer(Component, defaultProps) {
    return function(customProps) {
        const props = { ...defaultProps, ...customProps }
        return shallow(<Component {...props} />)
    }
}
