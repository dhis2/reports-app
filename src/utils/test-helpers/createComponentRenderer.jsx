import { shallow } from 'enzyme'
import React from 'react'

export default function createComponentRenderer(Component, defaultProps) {
    return function (customProps) {
        const props = { ...defaultProps, ...customProps }
        return shallow(<Component {...props} />)
    }
}
