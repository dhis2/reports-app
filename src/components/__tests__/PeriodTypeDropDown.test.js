import createComponentRenderer from '../../utils/test-helpers/createComponentRenderer'
import PeriodTypeDropDown from '../PeriodTypeDropDown'

jest.mock('@dhis2/d2-ui-core', () => ({
    DropDown: 'DropDown',
    PeriodPicker: 'PeriodPicker',
}))

describe('Test <PeriodTypeDropDown /> rendering:', () => {
    const defaultProps = {
        ready: true,
        loadingError: '',
    }

    const componentRenderer = createComponentRenderer(
        PeriodTypeDropDown,
        defaultProps
    )

    it('It should render a <DropDown /> when ready', () => {
        expect(componentRenderer()).toMatchSnapshot()
    })

    it('It should render a loading message when not ready', () => {
        expect(componentRenderer({ ready: false })).toMatchSnapshot()
    })

    it('It should render an error message when loadingError is present', () => {
        expect(componentRenderer({ loadingError: 'Oops' })).toMatchSnapshot()
    })
})
