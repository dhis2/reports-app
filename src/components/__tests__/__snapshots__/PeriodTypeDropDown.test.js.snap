// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PeriodTypeDropDown /> rendering: It should render a <DropDown /> when ready 1`] = `
<DropDown
  emptyLabel="Select Period Type"
  fullWidth={true}
  hintText="Select Period Type"
/>
`;

exports[`Test <PeriodTypeDropDown /> rendering: It should render a loading message when not ready 1`] = `
<span>
  Loading period types dropdown
  <style>
    
                        span {
                            color: #f44336;
                        }
                    
  </style>
</span>
`;
